bootstore/src/schemes/v0/peer.rs

```diff
@@ -448,6 +522,108 @@ impl Node {
+                            current_generation: current_gen,
+                        },
+                    ));
+                } else if current_gen == config.generation {
+                    warn!(
+                        self.log,
+                        concat!(
+                            "Not updating network config: generation ",
+                            "{} is current"
+                        ),
+                        current_gen
+                    );
+                } else {
+                    self.network_config = Some(config.clone());
+                    NetworkConfig::save(
+                        &self.log,
+                        self.config.network_config_ledger_paths.clone(),
+                        config,
+                    )
+                    .await;
+                    // Broadacst the updated config. We only broadcast
+                    // when we successfully update it so we don't trigger an
+                    // endless broadcast storm.
+                    self.broadcast_network_config(None).await;
+                    let _ = responder.send(Ok(()));
+                }
+            }
+            NodeApiRequest::GetNetworkConfig { responder } => {
+                let _ = responder.send(self.network_config.clone());
+            }
+        }
+    }
+
+    // After we have updated our network config, we should send it out to all
+    // peers, with the exception of the peer we received it from if this was not
+    // a local update.
+    async fn broadcast_network_config(
+        &mut self,
+        excluded_peer: Option<&Baseboard>,
+    ) {
+        // We only call this method when there has been an update. Otherwise we
+        // have an invariant violation due to programmer error and should panic.
+        let network_config = self.network_config.as_ref().unwrap();
+        info!(
+            self.log,
+            "Broadcasting network config with generation {}",
+            network_config.generation
+        );
+        for (id, handle) in self
+            .established_connections
+            .iter()
+            .filter(|(id, _)| Some(*id) != excluded_peer)
+        {
+            debug!(
+                self.log,
+                "Sending network config with generation {} to {id}",
+                network_config.generation
+            );
+            // Send the current network config
+            if let Err(e) = handle
+                .tx
+                .send(MainToConnMsg::Msg(Msg::NetworkConfig(
+                    network_config.clone(),
+                )))
+                .await
+            {
+                warn!(
+                    self.log,
+                    concat!(
+                        "Failed to send network config to connection ",
+                        "management task for {} {:?}"
+                    ),
+                    id,
+                    e
+                );
```

Comment:
Could probably put this in a method since it's identical in 3 places

Reply:
Done in [cd37295](https://github.com/oxidecomputer/omicron/pull/3612/commits/cd3729569e81ebd495999a73ef5ee06a03af3085)